src/ansi-c/c_typecheck_base.cpp

```diff
@@ -729,6 +729,7 @@ void c_typecheck_baset::typecheck_declaration(
         if(as_const(code_type).requires().is_not_nil())
         {
           auto &requires = code_type.requires();
+          typecheck_expr_requires(requires);
```

Comment:
The function call right below this line is `typecheck_expr(requires);`, so this is confusing.

Could you please rename the new function to `disallow_history_variables(requires)`? That would resolve the confusion and the name of your function would also match exactly what it's doing -- which isn't type checking.